### PR TITLE
feat: add Google Vertex AI as provider to utilize gemini via GCP

### PR DESCRIPTION
**Pull request by mfahlandt**
## 📑 Description
Adds VertexAI of GCP as a possible provider - this was asked as a possible feature here: https://k8sgpt.slack.com/archives/C050SKW7SHF/p1708033891149459

Another reason for adding it - currently Gemini API is not supported in a lot of regions (e.g. any EU Country + Australia) but Vertex AI is available to be used via GCP (https://ai.google.dev/available_regions) 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required [DocsPR](https://github.com/k8sgpt-ai/docs/pull/93)
- [x] All the tests have passed

### CODE REVIEW

**Review by AlexsJones — APPROVED** [review]

**Comment by arbreezy** [comment]
It lgtm but can't test it as I dont have vertex ai, it will be nice if you can share a screenshit of vertex integration while running k8sgpt analyze -e

**Review by JuHyung-Son — COMMENTED** [review]
overall looks good.
i just comment minor comments.

and does what if gemini API support other regions? then maybe we don't need to maintain this provider.

does it takes long time to gemini API supports other regions??

**Comment by mfahlandt** [comment]
> It lgtm but can't test it as I dont have vertex ai, it will be nice if you can share a screenshot of vertex integration while running k8sgpt analyze -e

Shure:
image

File: cmd/auth/add.go [hunk 1]
```diff
@@ -159,5 +160,7 @@ func init() {
 	// add flag for azure open ai engine/deployment name
 	addCmd.Flags().StringVarP(&engine, "engine", "e", "", "Azure AI deployment name (only for azureopenai backend)")
 	//add flag for amazonbedrock region name
-	addCmd.Flags().StringVarP(&providerRegion, "providerRegion", "r", "", "Provider Region name (only for amazonbedrock backend)")
+	addCmd.Flags().StringVarP(&providerRegion, "providerRegion", "r", "", "Provider Region name (only for amazonbedrock, googlevertexai backend)")
+	//add flag for vertexAI Project ID
+	addCmd.Flags().StringVarP(&providerId, "providerId", "i", "", "Provider Region name (only for googlevertexai backend)")
```
[the remark this answers is not in the document]
Comment (mfahlandt) [reply cmd/auth/add.go:165]: The provider ID is used in projectId := config.GetProviderId() in googlevertexai.go to retrieve the projectID, i named it providerid to be able to reuse it in other ai backends that might come up.

Reason for this is, vertex needs a specific google project where it is called in, this is for billing reasons - and you are right description need to be modified thx!

**Comment by mfahlandt** [comment]
> and does what if gemini API support other regions? then maybe we don't need to maintain this provider.
> 
> does it takes long time to gemini API supports other regions??

We could potentially then remove this provider. 
For now Vertex is the only option to use Gemini in GCP and this might still be the case after Gemini API expands, as the billing is different as it looks like. 
In this [Slack Thread](https://k8sgpt.slack.com/archives/C050SKW7SHF/p1708033891149459) the request was because the company does allow GCP but not Gemini Studio. 
I would treat it the same way as Bedrock & Sagemaker are treated for AWS.
The main issue is also for Gemini API this is more of a Law topic, as Gemini API is basically not supported in any region with extensive GDPR laws - e.g. EU/GB/Australia - my bet as a european this will take some time.

File: pkg/ai/googlevertexai.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+	"cloud.google.com/go/vertexai/genai"
+	"github.com/fatih/color"
+)
+
+const googleVertexAIClientName = "googlevertexai"
+
+type GoogleVertexAIClient struct {
+	client *genai.Client
+
```
[the remark this answers is not in the document]
Comment (mfahlandt) [reply pkg/ai/googlevertexai.go:29]: can you clarify? I tried to be in the same style as [the other google provider](https://github.com/k8sgpt-ai/k8sgpt/blob/main/pkg/ai/googlegenai.go#L29)

[the remark this answers is not in the document]
Comment (JuHyung-Son) [reply pkg/ai/googlevertexai.go:29]: oh i missed it. i wonder why there is blank line.

**Review by JuHyung-Son — APPROVED** [review]
overall good!

and need to resolve DCO

**Pull request merged**